### ISSUE TITLE
Exceptions instead of die.

### ISSUE THREAD

**Issue by bbkr** [mempko/PKafka#6]
Currently writing tests is hard because all errors ends as Ad::Hoc exceptions. Parsing error message is unreliable and a bad practice. Best practice in Perl 6 is to define exceptions in PKafka scope, for example:

```
class PKafka::X::ConsumeStart is Exception {
   has $.partition;
   has $.offset;
};
```

And in code:

```
my $res = PKafka::rd_kafka_consume_start($!topic, $partition, $offset);
PKafka::X::ConsumeStart->new( :$partition, :$offset, message => PKafka::errno2str() )->throw() if $res == -1;
```

It will allow to write good test suite for module and handle errors in application more gracefully.

**Comment by mempko** [comment]
Typed exceptions are definitely a good idea in any language. I used ad-hoc here temporarily.  I'll add them soon.

:+1:

**Comment by mempko** [comment]
@bbkr This has been resolved, see

https://github.com/mempko/PKafka/commit/42d59ee4f5b95469999f9aea52ca65a4b3e67934

**Comment by bbkr** [comment]
Awesome!

I found two tiny issues:

`PKafka::X::ConsumeError is Exception` - There is no need to use "Error" in class name, `PKafka::X::Consume` means the same. This is similar bad practice as for example naming tables "users_table" in databases.

Exception names are not consistent - there is `Consume` (infinitive verb) in one place but `ProducingMessage` (noun) instead of `ProduceMessage`or simply `Produce`  in another.

**Comment by bbkr** [comment]
Bug:

```
method message {"Error stopping consuming partition $.partition of topic $.topic: { PKafka::errno2str() }"}
```

Sub errno2str may not be evaluated immediately - exception object is created, thrown and catched. So you may get unrelated error while rethrowing it later. It should be stored while exception is created, probably in submethod BUILD.

**Comment by mempko** [comment]
Great catch on the bug. Should be fixed in this commit https://github.com/mempko/PKafka/commit/ade9f52a775b8f54b452c7e8cbd87c0aeda29119

I also updated the exception names, for example removing Error as you are right that it is redundant. 

I am keeping names like ProducingMessage vs ProduceMessage because it reads better. If the 'X' means "Error" then PKakfa::X::ProducingMessage reads like "Error Producing Message" vs "Error Produce Message"

I pushed other commits with consistent error names.

@bbkr
